pdns/pdns_recursor.cc

```diff
@@ -1284,9 +1301,8 @@ static void startDoResolve(void *p)
          OPT record.  This MUST also occur when a truncated response (using
          the DNS header's TC bit) is returned."
       */
-      if (addRecordToPacket(pw, makeOpt(edo.d_packetsize, 0, edo.d_Z), minTTL, dc->d_ttlCap, maxanswersize)) {
-        pw.commit();
-      }
+      pw.addOpt(g_udpTruncationThreshold, ednsExtRCode, DNSSECOK ? EDNSOpts::DNSSECOK : 0, returnedEdnsOptions);
```

Comment:
`handleNewUDPQuestion` accepts queries of up to 256 bytes, not `udp-truncation-threshold`

Reply:
I lied, it's 1500 (see #6627). 256 would be a fine limit, though!

Reply:
256 is forbidden by RFC 6891 section 6.2.5:

>    Values of less than 512 bytes MUST be treated as equal to 512 bytes.

(ok, not forbidden technically, but ignored...). So I've set this to 512.